### PR TITLE
Implement Translator 

### PR DESCRIPTION
**Pull request by vipulgupta2048**
Clean new PR, ported just one test to PyTest to get a view of how we are going forward. 
Reviews requested @rennerocha @ejulio 

Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipulgupta2048@gmail.com>

### CODE REVIEW

File: tests/test_validators_cerberus.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
+from __future__ import absolute_import
+import pytest
+
+from spidermon.contrib.validation import CerberusValidator
+from spidermon.contrib.validation import messages
+
+# Format for parameters
```
Comment (rennerocha) [inline tests/test_validators_cerberus.py]: This comment is unnecessary. The name of the params should be self-explanatory.

File: tests/test_validators_cerberus.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+# 1. schema
+# 2. data
+# 3. valid
+# 4. expected_errors
+# 5. Id of test
+
+# Format of test
```
Comment (rennerocha) [inline tests/test_validators_cerberus.py]: Here you are just copying the implementation code. Remove this comment. The parameters and the code is self-explanatory. No need for this kind of comment here.

**Review by rennerocha — CHANGES_REQUESTED** [review]
Left some comments. Please review them.

File: tests/test_validators_cerberus.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        pytest.param(
+            {"foo": {}},
+            {},
+            True,
+            {},
+            id="NOT REQUIRED - by default"
+        )
+    ]
+)
+def test_required_rule(schema, data, valid, expected_errors):
```
Comment (ejulio) [inline tests/test_validators_cerberus.py]: I'm not sure if messages should be validated through the validator.
Probably you should test the translator directly here.
Then, these messages should be asserted in the validator tests.

@rennerocha , any thoughts?

**Review by ejulio — CHANGES_REQUESTED** [review]
@vipulgupta2048 , please remove commented code before creating the PR.
If we need to go back for some reason, git allows us to do that.
Also, no need to mark the PR with _test/no tests_, they should always have tests when required

**Comment by vipulgupta2048** [comment]
Removed all commented code :heavy_check_mark:

**Review by vipulgupta2048 — COMMENTED** [review]
Ready to review :rocket: 
I feel the tests are looking quite robust, as I tried to play best of both world. 
I tried to implement the advice of @rennerocha  to have more diversified tests, hence this time these are more a bit more complex, but tests quite a lot of things. Following the pattern of not trying to write more tests, but have smarter tests instead. 

I also implemented @ejulio advice regarding testing of the translator as a component or unit.  Hence, instantiated the class and tried to have some good tests done on them. I have also added integration tests, on new messages that I have added for my peace of mind to see that they are in fact working. 

Let me know you thoughts, looking forward to the review.

File: tests/test_validators_cerberus.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            "Unknown field",
+            messages.UNEXPECTED_FIELD,
+            id="Invalid error of UNEXPECTED FIELD",
+        ),
+    ],
+)
+def test_cerberus_translator_invalid_messages(error, message):
+    translator = CerberusMessageTranslator()
+    with pytest.raises(AssertionError):
+        assert translator.translate_message(error) == (message)
```
Comment (ejulio) [inline tests/test_validators_cerberus.py]: no need for `()` around `message`

File: tests/test_validators_cerberus.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+import pytest
+
+from spidermon.contrib.validation import CerberusValidator
+from spidermon.contrib.validation import CerberusMessageTranslator
+from spidermon.contrib.validation import messages
+
+
+@pytest.mark.parametrize(
```
Comment (ejulio) [inline tests/test_validators_cerberus.py]: maybe these tests should be in its own file `test_cerberus_translator.py`

File: tests/test_validators_cerberus.py (hunk 4 above)
Comment (vipulgupta2048) [reply tests/test_validators_cerberus.py]: Gotcha!

**Comment by vipulgupta2048** [comment]
@ejulio @rennerocha Branch updated, changes implemented.

File: tests/test_cerberus_translator.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                "1": {"empty": False, "type": "string"},
+            },
+            {"foo": {"address": "my address", "city": "my town"}, "1": ""},
+            False,
+            {"1": [messages.EMPTY_NOT_ALLOWED]},
+            id="Empty values - Invalid Case",
+        ),
+    ],
+)
+def test_required_rule(schema, data, valid, expected_errors):
```
Comment (ejulio) [inline tests/test_cerberus_translator.py:100]: This test shouldn't be here, right?

Comment (vipulgupta2048) [reply tests/test_cerberus_translator.py:100]: Yes @ejulio as per our previous discussion. I took some time to learn about testing concepts, tools, and best practices. I feel here in this instance these tests (Integration tests) can go well with our unit tests just because of the fact that I am testing only those messages which are added new to the `messages.py`, specific to Spidermon, and helps us debug faster when these edge-cases so as to say breaks somewhere down the line.  

These new tests actually helped me out, to debug and write better messages for the translator as well as find that `feature we have missed` that was throwing a `TypeError` (Also talked about this)

For the location, I know well now that this isn't the right spot for these tests considering their nature. But, I think they will be good here as they cover new messages, also everything related to the translator that might potentially trouble us is in just one spot and I think we can grow quite better from here on out. 

**Here's my flow**
If we add a new message to the translator, we can also add a test to test it out below. Leading to Unit test and integration going hand in hand rather than us switching files to find 2 different tests that more or less are made for the same component. What say?

**Review by ejulio — APPROVED** [review]
LGTM!
@rennerocha , what are your thoughts?
